plugins/catalog-react/src/hooks/useEntityPermission.test.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { catalogEntityDeletePermission } from '@backstage/plugin-catalog-common';
+import { renderHook } from '@testing-library/react-hooks';
+import { useEntityPermission } from './useEntityPermission';
+import { useEntity } from './useEntity';
+import { usePermission } from '@backstage/plugin-permission-react';
+
+jest.mock('./useEntity', () => ({
+  ...jest.requireActual('./useEntity'),
+  useEntity: jest.fn(),
+}));
+jest.mock('@backstage/plugin-permission-react', () => ({
+  ...jest.requireActual('@backstage/plugin-permission-react'),
+  usePermission: jest.fn(),
+}));
+const useEntityMock = useEntity as jest.Mock;
+const usePermissionMock = usePermission as jest.Mock;
```

Comment:
The most recent refactor of the hook meant moving away from using the provided mock of the api, in order to tune the return values of the underlying hooks to test each specific case in isolation.

Reply:
Is this really necessary, or only because you wanted to shorten the test setup code? I'd have preferred the longer and more explicit setup personally :)